docs/agent_container_manual_deploy.md

```diff
@@ -0,0 +1,76 @@
+ * HZN_AGENT_IMAGE_TAG=<version of the agent container to use>
+ * CONFIG_FILE=<Location of the config file>
+ 	* The config file with the contents from above.
+ * HZN_MGMT_HUB_CERT_MOUNT="-v <ssl cert file on host>:<$HZN_MGMT_HUB_CERT_PATH>"
+ 	* This needs to be set in the config file so the agent can find the cert after it starts.
+ * DOCKER_NAME=<name for the agent container>
+ * HORIZON_AGENT_PORT=<port number>
+ 	* The port to expose from the container that hzn will call the agent on. Typically 8081.
+ * DOCKER_ADD_HOSTS="--add-host=$<host name to add to container hosts file>"
+ 	* Omit if not needed
+
+Example:
+```bash
+HZN_AGENT_IMAGE=openhorizon/amd64_anax
+HZN_AGENT_IMAGE_TAG=latest
+CONFIG_FILE=/etc/default/horizon
+HZN_MGMT_HUB_CERT_PATH=/etc/default/horizon/agent.crt
+DOCKER_NAME=horizon1
+HORIZON_AGENT_PORT=8081
+```
+
+```bash
+docker run $DOCKER_ADD_HOSTS -d --restart always --name $DOCKER_NAME --privileged -p 127.0.0.1:$HORIZON_AGENT_PORT:8510 -e DOCKER_NAME=$DOCKER_NAME -v /var/run/docker.sock:/var/run/docker.sock -v $CONFIG_FILE:/etc/default/horizon:ro $HZN_MGMT_HUB_CERT_MOUNT -v $DOCKER_NAME_var:/var/horizon/ -v $DOCKER_NAME_etc:/etc/horizon/ -v $DOCKER_NAME_fss:/var/tmp/horizon/$DOCKER_NAME $HZN_AGENT_IMAGE:$HZN_AGENT_IMAGE_TAG
+
+export HORIZON_URL=http://localhost:$HORIZON_AGENT_PORT
+```
+
+
+The docker run command uses the following arguments. Here is what they are for:
+ * "--name $DOCKER_NAME"
+ 	* Names the container something meaningful to the user for convenience.
+ * "--add-host=$HZN_EXCHANGE_HOSTS"
+ 	* This allows the user to add the exchange to the containers host list.
+ * "-p 127.0.0.1:$HORIZON_AGENT_PORT:$ANAX_AGENT_PORT"
+ 	* This exposes a port that `hzn` can reach the agent on.
+ * "-e DOCKER_NAME=$DOCKER_NAME"
+ 	* Lets the agent container know what its name is.
+ * "-v /var/run/docker.sock:/var/run/docker.sock"
+ 	* Gives the container access to the docker socket so it can control service containers.
+ * "-v $CONFIG_FILE_MOUNT:/etc/default/horizon:ro"
+ 	* The config file lets the user set variables to tell the agent about itself and the exchange it is to work with.
+ * "-v $HZN_MGMT_HUB_CERT_PATH:$HZN_MGMT_HUB_CERT_PATH"
+ 	* Maps the ssl certificate for the exchange into the agent container. The agent will look for the cert wherever `HZN_MGMT_HUB_CERT_PATH` is set to in the config file.
+ * "-v $DOCKER_NAME_var:/var/horizon/"
+ 	* This volume will hold the database for the agent.
+ * "-v $DOCKER_NAME_etc:/etc/horizon/"
+	* This contains some initial configuration information for the agent.
+	* This file in the container image is non-empty so mounting it to a non-empty host folder will cause the agent to fail.
+ * "-v $DOCKER_NAME_fss:/var/tmp/horizon/horizon1"
+ 	* This volume is used by the edge sync service for storing models downloaded by the service.
+ * "$HZN_AGENT_IMAGE:$HZN_AGENT_IMAGE_TAG"
+	* The agent container image that will be started.
```

Comment:
Just realized, I think we need a blurb about registering the node. Some of the choices made above and variables are relevant, so we need to show a register command also. For example, HZN_EXCHANGE_USER_AUTH isnt used until the register.

Reply:
Added at the bottom